validator/src/types.rs

```diff
@@ -49,6 +59,39 @@ impl std::error::Error for Errors {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct Error {
```

Comment:
It clashes with `std::error::Error`, at least in my mind. But after thinking about it, it's not that bad